modules/navier_stokes/test/tests/finite_volume/ins/channel-flow/2d-rc-no-slip.i

```diff
@@ -131,11 +131,75 @@ velocity_interp_method = 'rc'
 [Executioner]
   type = Steady
   solve_type = 'NEWTON'
-  petsc_options_iname = '-pc_type -pc_factor_shift_type'
-  petsc_options_value = 'lu NONZERO'
   nl_rel_tol = 1e-12
 []
 
+[Preconditioning]
+  active = SMP
+  [FSP]
+    type = FSP
+    # It is the starting point of splitting
+    topsplit = 'up' # 'up' should match the following block name
+    [up]
+      splitting = 'u p' # 'u' and 'p' are the names of subsolvers
+      splitting_type  = schur
+      # Splitting type is set as schur, because the pressure part of Stokes-like systems
+      # is not diagonally dominant. CAN NOT use additive, multiplicative and etc.
+      # Original system:
+      # | A B | | u | = | f_u |
+      # | C 0 | | p |   | f_v |
+      # is factorized into
+      # |I        0 | | A    0|  | I  A^{-1}B | | u | = | f_u |
+      # |CA^{-1}  I | | 0   -S|  | 0    I     | | p |   | f_v |
+      # S = CA^{-1}B
+      # The preconditioning is accomplished via the following steps
+      # (1) p^{(0)} = f_v - CA^{-1}f_u,
+      # (2) pressure = (-S)^{-1} p^{(0)}
+      # (3) u = A^{-1}(f_u-Bp)
+      petsc_options_iname = '-pc_fieldsplit_schur_fact_type  -pc_fieldsplit_schur_precondition'
+      petsc_options_value = 'full                            selfp'
+      # Factorization type here is full, which means we approximate the original system
+      # exactly. There are three other options:
+      # diag:
+      # | A    0|
+      # | 0   -S|
+      # lower:
+      # |I        0  |
+      # |CA^{-1}  -S |
+      # upper:
+      # | I  A^{-1}B |
+      # | 0    -S    |
+      # The preconditioning matrix is set as selfp, which means we explicitly form a
+      # matrix \hat{S} = C(diag(A))^{-1}B. We do not compute the inverse of A, but instead, we compute
+      # the inverse of diag(A).
+    []
+    [u]
+      vars = 'vel_x vel_y'
+      # PETSc options for this subsolver
+      # A prefix will be applied, so just put the options for this subsolver only
+      petsc_options_iname = '-pc_type -pc_hypre_type -ksp_type -ksp_rtol'
+      petsc_options_value = 'hypre    boomeramg      gmres     1e-4'
```

Comment:
btw I m not sure if this is on purpose, but on the other field split we set it to precondition only, with gmres here as the ksp_type I think it solves it